schema/string_test.go

```diff
@@ -3,36 +3,57 @@ package schema
 import "testing"
 
 // To be in par with the python library.
-func TestCastString_URIMustRequireScheme(t *testing.T) {
-	if _, err := castString(stringURI, "google.com"); err == nil {
+func TestDecodeString_URIMustRequireScheme(t *testing.T) {
+	if _, err := decodeString(stringURI, "google.com", Constraints{}); err == nil {
 		t.Errorf("want:err got:nil")
 	}
 }
 
-func TestCastString_InvalidUUIDVersion(t *testing.T) {
+func TestDecodeString_InvalidUUIDVersion(t *testing.T) {
 	// This is a uuid3: namespace DNS and python.org.
-	if _, err := castString(stringUUID, "6fa459ea-ee8a-3ca4-894e-db77e160355e"); err == nil {
+	if _, err := decodeString(stringUUID, "6fa459ea-ee8a-3ca4-894e-db77e160355e", Constraints{}); err == nil {
 		t.Errorf("want:err got:nil")
 	}
 }
 
-func TestCastString_Success(t *testing.T) {
-	var data = []struct {
-		Desc   string
-		Value  string
-		Format string
-	}{
-		{"URI", "http://google.com", stringURI},
-		{"Email", "foo@bar.com", stringEmail},
-		{"UUID", "C56A4180-65AA-42EC-A945-5FD21DEC0538", stringUUID},
-	}
-	for _, d := range data {
-		v, err := castString(d.Format, d.Value)
-		if err != nil {
-			t.Errorf("want:nil got:%q", err)
+func TestDecodeString_Success(t *testing.T) {
+	t.Run("ValidMinLength", func(t *testing.T) {
```

Comment:
I believe there is no need for this subtest, as the test function is already called TestDecodeString_Success. You could update this test adding more cases for other constraints. What do you think?

Reply:
You're right, We should focus on "Success" testing on current `TestDecodeString_Success` function.
Maybe I can create the other function call TestDecodeString_Error and testing error cases it that function.

Reply:
Maybe TestDecodeString_ErrorCheckingConstraints ? (which is more likely what you would be focusing)

Reply:
That's good.
I can follow this [way](https://github.com/frictionlessdata/tableschema-go/blob/master/schema/time_test.go#L32) to handle different constraints testing cases.

Reply:
Yeah. That is super nice.